src/App.js

```diff
@@ -1,25 +1,155 @@
 import React, { Component } from 'react';
-import { Text, View } from 'react-native';
+import { Text, View, AppState } from 'react-native';
 import { Actions } from 'react-native-router-flux';
+import PushController from './components/PushController';
+import PushNotification from 'react-native-push-notification';
 import { Spinner, Button } from './components/common';
+// import BackgroundJob from 'react-native-background-job';
+import BackgroundTask from 'react-native-background-task';
+
 // import { PermissionsAndroid } from 'react-native';
 import utils from './utils/methods';
-import defaultState from './utils/constants';
+
+// const firstJobKey = "firstJobKey";
+//onst regularJobKey = "regularJobKey";
+BackgroundTask.define(async () => {
+  const ID = 'ID: ' + Math.random().toFixed(4);
+  console.log(`1- ${ID} - ${new Date()} Background Job fired!.`);
+
+  BackgroundTask.cancel(); // ios/android
+
+  const decision = await utils.getCachedItems();
+
+  PushNotification.localNotification({
+    title: "",
+    message: `weather: ${decision.weather.main.temp}
+    ${decision.weather.weather[0].main}
+    ${new Date()}`, // (required)
+    playSound: false,
+  });
+
+  BackgroundTask.schedule({
+      period: 900, //24 hrs
+  });
+  BackgroundTask.finish();
+});
+
+// BackgroundJob.register({
+//   jobKey: regularJobKey,
+//   job: async () => {
+//     console.log(`${new Date()} Background Job fired!. Key = ${regularJobKey}`);
+//     console.log("fetching weather");
+//
+//     const decision = await utils.getCachedItems();
+//     console.log("got weather");
+//     PushNotification.localNotification({
+//       message: `weather: ${decision.weather.main.temp}
+//       ${decision.weather.weather[0].main}`, // (required)
+//       playSound: false,
+//     });
+//     console.log("sent notification");
+//     }
+// });
+//
+// BackgroundJob.register({
+//   jobKey: firstJobKey,
+//   job: () => {
+//     //App.setNotification(App.fetchWeather());
+//     console.log(`${new Date()} Background Job fired!. Key = ${firstJobKey}`)
+//     //if (PushNotification) PushNotification.cancelAllLocalNotifications();
+//     BackgroundJob.cancelAll();
+//     BackgroundJob.schedule({
+//       jobKey: regularJobKey,
+//       period: 100000, //5mins
+//       //period: 86400000, //24hrs
+//       //networkType: 1,
+//     });
+//     }
+// });
 
 export default class App extends Component<{}> {
   // Default values
-  state = { ...defaultState }
+ constructor(props) {
+   super(props);
+   this.handleAppStateChange = this.handleAppStateChange.bind(this);
+   this.state = {
+      isFetching:false,
+      isRaining: false,
+      remark:'',
+      /* localStorage */
+      position: null,
+      weather: null,
+      lastUpdated: null,
+      /* SettingStorage */
+      reminderOn: false,
+      date: new Date('2017-01-01T07:00:00.000Z'),
```

Comment:
do we need this string inside? new Date()?

Reply:
yep